programs/uxd/src/instructions/credix_lp/exchange_liquidity_with_credix_lp_depository.rs

```diff
@@ -0,0 +1,334 @@
+use anchor_lang::prelude::*;
+use anchor_spl::token;
+use anchor_spl::token::Mint;
+use anchor_spl::token::Token;
+use anchor_spl::token::TokenAccount;
+use anchor_spl::token::Transfer;
+
+use crate::error::UxdError;
+use crate::events::ExchangeLiquidityWithCredixLpDepositoryEvent;
+use crate::state::controller::Controller;
+use crate::state::credix_lp_depository::CredixLpDepository;
+use crate::state::identity_depository::IdentityDepository;
+use crate::utils::checked_add;
+use crate::utils::checked_as_u64;
+use crate::utils::checked_sub;
+use crate::utils::compute_decrease;
+use crate::utils::compute_increase;
+use crate::validate_is_program_frozen;
+use crate::CONTROLLER_NAMESPACE;
+use crate::CREDIX_LP_DEPOSITORY_NAMESPACE;
+use crate::IDENTITY_DEPOSITORY_COLLATERAL_NAMESPACE;
+use crate::IDENTITY_DEPOSITORY_NAMESPACE;
+
+#[derive(Accounts)]
+pub struct ExchangeLiquidityWithCredixLpDepository<'info> {
```

Comment:
So basically, this ix allow any user to buy Credix shares at the rate of:

SHARES OWNED / REDEEMABLE UNDER MANAGEMENT

Buying credix shares at premium rate

Reply:
Yep! exactly ser, the user can take on the illiquidity risk instead of the protocol at any time basically.